lib/puma/cluster.rb

```diff
@@ -365,13 +365,15 @@ def phased_restart
     def stop
       @status = :stop
       wakeup!
+      @launcher.events.fire_on_stopped!
```

Comment:
I am not sure if I made the right thing here

Reply:
Hmmm I'm not sure either, I have to look closer at when we use each method.